freeplane/src/main/java/org/freeplane/features/styles/ConditionalStyleModel.java

```diff
@@ -200,21 +210,70 @@ void swapConditions(int index1, int index2){
 		table.fireTableRowsUpdated(index1, index1);
 		table.fireTableRowsUpdated(index2, index2);
 	}
-	
+
 	void moveUp(int index){
 		if(index == 0){
 			return;
 		}
 		swapConditions(index, index - 1);
 	}
-	
+
 	void moveDown(int index){
 		if(index == styles.size() - 1){
 			return;
 		}
 		swapConditions(index, index + 1);
 	}
-	
+
+	void move(int index, int toIndex) {
+		int lastIndex = styles.size() - 1;
+		if (toIndex < 0 || toIndex > lastIndex)
+			throw new IndexOutOfBoundsException(String.format("toIndex (%d) is out of bounds - expected between 0 and %d", toIndex, lastIndex));
+		final Item itemAtIndex = styles.remove(index);
+		styles.add(toIndex, itemAtIndex);
+		if (table == null) {
+			return;
+		}
+		table.fireTableRowsUpdated(index, index);
```

Comment:
moving row from index 1 to index 5 you actually change all rows between 1 and 5 because rows 2 3, 4 and 5 go up

Reply:
A good catch. Thanks :+1: